File: app/src/main/kotlin/com/egecius/gtdx/utils/Sorter.java
package com.egecius.gtdx.utils;

import com.egecius.gtdx.datatypes.TodoTask;

import java.util.List;

/**
 * Sorts collections
 */
public interface Sorter {

	/** Returns sorted list of to-do tasks, with newest being at the top */
	List<TodoTask> sortNewestFirst(List<TodoTask> unsorted);

}

File: app/src/main/kotlin/com/egecius/gtdx/shared/utils/Sorter.kt
package com.egecius.gtdx.shared.utils

import com.egecius.gtdx.datatypes.TodoTask

/**
 * Sorts collections
 */
interface Sorter {

    /** Returns sorted list of to-do tasks, with newest being at the top  */
    fun sortNewestFirst(unsorted: List<TodoTask>): List<TodoTask>

}
